### DISCUSSION TITLE
Using DDP for training VITS

### DISCUSSION THREAD

**Discussion by OmiWakode** [coqui-ai/TTS#1476] [General Q&A] [answered]
Hey,
As I was going to the [VITS](https://github.com/coqui-ai/TTS/blob/0cf3265a4686d7e856bd472cdaf1572d61cab2b8/TTS/tts/models/vits.py#L1369) code, I saw some usage of `DistributedDataSampler`. I see that the sampler is being initialized but not passed to `DistributedDataLoader` as a sampler argument. How does this ensure that each process gets a different slice of the training data, whenever loading the data by `DistributedDataLoader`.

**Comment by erogol** [comment]
Good point, I think it's a bug

**Comment by Edresson** [comment, marked as answer]
Hi @OmiWakode ,  this bug was fixed in PR #1216 . Until the new version is released, you can use the dev branch to use the DDP in VITS model training.

**Reply by OmiWakode** [reply]
Hey. Can you be more precise on how to do that? dev branch doesn't have a distribute.py file.

**Reply by Edresson** [reply]
The distribute.py script was moved to the [Trainer package](https://github.com/coqui-ai/Trainer). You can use the distribute script with the following command:
`python3 -m trainer.distribute --script `
